Handle addition of ID to NVT preferences.

kroosec opened the pull request.

cfi-gb commented on src/manage.c:
```diff
@@ -2139,7 +2127,7 @@ send_alive_test_preferences (target_t target)
     {
       /* Also select a method, otherwise Ping Host logs a warning. */
-      if (sendf_to_server (OID_PING_HOST ":checkbox:Do a TCP ping <|> yes\n"))
+      if (sendf_to_server (OID_PING_HOST ":1:checkbox:Do a TCP ping <|> yes\n"))
```
The ids where added to ping_host.nasl today.

cfi-gb commented on src/manage.c:
```diff
@@ -5393,7 +5382,7 @@ run_otp_task (task_t task, scanner_t scanner, int from, char **report_id)
                                       (&credentials)));
 
-              if (sendf_to_server (OID_SSH_AUTH ":file:SSH private key:"
+              if (sendf_to_server (OID_SSH_AUTH ":4:file:SSH private key:"
```
The ids where added to ssh_authorization_init.nasl today.

cfi-gb commented on src/manage.c:
```diff
@@ -5412,9 +5401,9 @@ run_otp_task (task_t task, scanner_t scanner, int from, char **report_id)
+          if (sendf_to_server (OID_SMB_AUTH ":1:entry:SMB login: <|> %s\n",
                                user ? user : "")
-              || sendf_to_server (OID_SMB_AUTH ":password:SMB password:"
+              || sendf_to_server (OID_SMB_AUTH ":2:password:SMB password:"
```
The ids where added to smb_authorization.nasl today.

cfi-gb commented on src/manage.c:
```diff
@@ -5443,10 +5432,10 @@ run_otp_task (task_t task, scanner_t scanner, int from, char **report_id)
                                " <|> %s\n",
                                user ? user : "")
-              || sendf_to_server (OID_ESXI_AUTH ":password:ESXi login password:"
+              || sendf_to_server (OID_ESXI_AUTH ":2:password:ESXi login password:"
```
The ids where added to gb_esxi_authorization.nasl today.

cfi-gb commented on src/manage.c:
```diff
@@ -5482,25 +5471,25 @@ run_otp_task (task_t task, scanner_t scanner, int from, char **report_id)
                                   " <|> %s\n",
                                   privacy_password ? privacy_password : "")
-              || sendf_to_server (OID_SNMP_AUTH ":radio:"
+              || sendf_to_server (OID_SNMP_AUTH ":6:radio:"
```
The ids where added to gb_snmp_authorization.nasl today.

kroosec commented on src/gmp.c:
```diff
@@ -749,8 +752,9 @@ create_config_data_reset (create_config_data_t *data)
+  g_free (import->preference_id);
   g_free (import->preference_name);
-  g_free (import->preference_name);
+  g_free (import->preference_hr_name);
```
Moved to a separate commit.

kroosec commented on src/gmp.c:
```diff
@@ -3098,23 +3103,24 @@ modify_auth_data_reset (modify_auth_data_t * data)
 {
-  free (data->comment);
-  free (data->config_id);
+  g_free (data->comment);
```
Removed the change. I agree that it's better to use a different PR to change all parts together.

kroosec commented on src/manage.c:
```diff
@@ -2090,42 +2090,34 @@ send_alive_test_preferences (target_t target)
                         : "no"))
     return -1;
 
-  if (sendf_to_server (OID_PING_HOST ":checkbox:TCP ping tries only TCP-SYN ping"
```
Fixed. Seemed like a duplicate. Added an ID to it too.

kroosec commented on src/gmp.c:
```diff
@@ -10120,25 +10132,28 @@ void
   buffer_xml_append_printf (buffer,
                             "<preference>"
                             "<nvt oid=\"%s\"><name>%s</name></nvt>"
+                            "<id>%s</id>"
```
Added GMP documentation and compatibility changes section.

kroosec commented:
> I'm keen to get this in but would like to see the GMP doc updated with the ID, and also a check of why the OID_PING_HOST preference was removed.

Fixed all issues. Also added to the PR a routine to update old-style preferences in config_preferences.

cfi-gb commented on src/manage.c:
```diff
@@ -2094,42 +2090,42 @@ send_alive_test_preferences (target_t target)
     return -1;
 
-  if (sendf_to_server (OID_PING_HOST ":checkbox:TCP ping tries only TCP-SYN ping"
+  if (sendf_to_server (OID_PING_HOST ":7:checkbox:TCP ping tries only TCP-SYN ping"
```
The id:7 for this preference was added to ping_host.nasl today.

cfi-gb commented on src/manage_config_host_discovery.c:
```diff
@@ -122,14 +122,14 @@ check_config_host_discovery (const char *uuid)
            " VALUES ((SELECT id FROM configs WHERE uuid = '%s'),"
            "         'PLUGINS_PREFS',"
-           "         '" OID_GLOBAL_SETTINGS ":checkbox:Strictly unauthenticated',"
+           "         '" OID_GLOBAL_SETTINGS ":1:checkbox:Strictly unauthenticated',"
```
The id:1 for this preference was added to global_settings.nasl today.
